core/common/src/main/java/alluxio/conf/path/PrefixPathMatcher.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf.path;
+
+import alluxio.AlluxioURI;
+
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Matched patterns are prefixes of a path, the path must be in the subtree of the prefixes.
```

Comment:
can we replace "prefixes" with "paths"? They are just paths, right?

Reply:
rephrased to `Patterns to be matched against are paths, a path matches a pattern if it is the same as the pattern or in the subtree of the pattern.`

Reply:
why is it necessary to introduce the terminology "pattern"? I feel like it makes the documentation more abstract and harder to understand

Reply:
make sense, will update related docs.